lib/charms/kubernetes_charm_libraries/v0/kubernetes_volumes_patch.py

```diff
@@ -0,0 +1,618 @@
+        additional_volumemounts = self._generate_volumemounts_from_requested_hugepage()
+        current_volumemounts = self.kubernetes.list_volumemounts(
+            statefulset_name=self.model.app.name, container_name=self.container_name
+        )
+        for current_volumemount in current_volumemounts:
+            if not self._volumemount_is_hugepages(current_volumemount):
+                additional_volumemounts.append(current_volumemount)
+
+        # handle statefulset volumes
+        additional_volumes = self._generate_volumes_from_requested_hugepage()
+        current_volumes = self.kubernetes.list_volumes(
+            statefulset_name=self.model.app.name,
+        )
+        for current_volume in current_volumes:
+            if not self._volume_is_hugepages(current_volume):
+                additional_volumes.append(current_volume)
+
+        # handle resource requirements
+        additional_resources = self._generate_resource_requirements_from_requested_hugepage()
+        current_resources = self.kubernetes.list_container_resources(
+            statefulset_name=self.model.app.name, container_name=self.container_name
+        )
+
+        if self.hugepages_volumes_func():
+            if current_resources.limits:
+                new_limits = {
+                    limit: value
+                    for limit, value in current_resources.limits.items()
+                    if not self._limit_or_resource_is_hugepages(limit)
+                }
+                new_limits = dict(new_limits.items() | additional_resources.limits.items())
+            else:
+                new_limits = additional_resources.limits
+            if current_resources.requests:
+                new_requests = {
+                    request: value
+                    for request, value in current_resources.requests.items()
+                    if not self._limit_or_resource_is_hugepages(request)
+                }
+                new_requests = dict(new_requests.items() | additional_resources.requests.items())
+            else:
+                new_requests = additional_resources.requests
+        else:
+            if current_resources.limits:
+                new_limits = {
+                    limit: value
+                    for limit, value in current_resources.limits.items()
+                    if not self._limit_or_resource_is_hugepages(limit)
+                }
+            else:
+                new_limits = current_resources.limits
+            if current_resources.requests:
+                new_requests = {
+                    request: value
+                    for request, value in current_resources.requests.items()
+                    if not self._limit_or_resource_is_hugepages(request)
+                }
+            else:
+                new_requests = current_resources.requests
+        new_resources = ResourceRequirements(
+            limits=new_limits, requests=new_requests, claims=current_resources.claims
```

Comment:
Too many if else blocks. I suggest to refactor this by defining some small methods to increase readability.

Reply:
I agree